### PR TITLE
Rearrange silver

### PR DESCRIPTION
**Pull request by bqi343**
moved two pointers before more on ordered sets

also fixed some minor issues in gold

### CODE REVIEW

File: content/3_Silver/Binary_Search_Ans.mdx [hunk 1]
```diff
@@ -230,10 +245,10 @@ int lstTrue(function<bool(int)> f, int lo, int hi) {
 }
 
 int main() {
-	cout << lstTrue([](int x) { return true; },2,10) << "\n"; // 10
+	cout << lstTrue(['Binary Search'](int x) { return true; },2,10) << "\n"; // 10
 	// all numbers satisfy the condition
-	cout << lstTrue([](int x) { return x*x <= 30; },2,10) << "\n"; // 5
-	cout << lstTrue([](int x) { return false; },2,10) << "\n"; // 1
+	cout << lstTrue(['Binary Search'](int x) { return x*x <= 30; },2,10) << "\n"; // 5
+	cout << lstTrue(['Binary Search'](int x) { return false; },2,10) << "\n"; // 1
```
Comment (frodakcin) [inline content/3_Silver/Binary_Search_Ans.mdx:251]: I assume these weren't intended?

File: content/3_Silver/Binary_Search_Ans.mdx [hunk 2]
```diff
@@ -273,7 +288,7 @@ public class test{
 		// if res is lo-1 that means no value in the range satisfies the condition
 		return res;
 	}
-	public static void main(String[] args) throws IOException
+	public static void main(String['Binary Search'] args) throws IOException
```
Comment (frodakcin) [inline content/3_Silver/Binary_Search_Ans.mdx:291]:
```suggestion
	public static void main(String[] args) throws IOException
```

File: content/3_Silver/Binary_Search_Ans.mdx [hunk 3]
```diff
@@ -458,9 +473,9 @@ int fstTrue(function<bool(int)> f, int lo, int hi) {
 }
 
 int main() {
-	cout << fstTrue([](int x) { return true; },2,10) << "\n"; // 2
-	cout << fstTrue([](int x) { return x*x >= 30; },2,10) << "\n"; // 6
-	cout << fstTrue([](int x) { return false; },2,10) << "\n"; // 11
+	cout << fstTrue(['Binary Search'](int x) { return true; },2,10) << "\n"; // 2
+	cout << fstTrue(['Binary Search'](int x) { return x*x >= 30; },2,10) << "\n"; // 6
+	cout << fstTrue(['Binary Search'](int x) { return false; },2,10) << "\n"; // 11
```
Comment (frodakcin) [inline content/3_Silver/Binary_Search_Ans.mdx:478]: Again here

File: content/3_Silver/Binary_Search_Ans.mdx [hunk 4]
```diff
@@ -486,7 +501,7 @@ public class test{
 		}
 		return lo;
 	}
-	public static void main(String[] args) throws IOException
+	public static void main(String['Binary Search'] args) throws IOException
```
Comment (frodakcin) [inline content/3_Silver/Binary_Search_Ans.mdx:504]: Here

File: content/3_Silver/Binary_Search_Ans.mdx [hunk 5]
```diff
@@ -587,8 +602,8 @@ int main() {
 ```java
 static int n;
 static long k;
-static long[] arr;
-public static void main(String[] args) {
+static long['Binary Search'] arr;
+public static void main(String['Binary Search'] args) {
```
Comment (frodakcin) [inline content/3_Silver/Binary_Search_Ans.mdx:606]: 2 more here

File: content/3_Silver/Binary_Search_Ans.mdx [hunk 6]
```diff
@@ -649,7 +664,7 @@ def search():
 		a //= 2
     return int(pos)
 
-v=[]
+v=['Binary Search']
```
Comment (frodakcin) [inline content/3_Silver/Binary_Search_Ans.mdx:667]: here

File: content/3_Silver/Binary_Search_Ans.mdx [hunk 7]
```diff
@@ -698,8 +713,8 @@ The `fstTrue` code given above does not necessarily work if `lo` is negative! Co
 
 ```cpp
 int main() {
-	cout << fstTrue([](int x) { return false; },-10,-10) << "\n"; // -8, should be -9
-	cout << fstTrue([](int x) { return true; },-10,-10) << "\n"; // infinite loop
+	cout << fstTrue(['Binary Search'](int x) { return false; },-10,-10) << "\n"; // -8, should be -9
+	cout << fstTrue(['Binary Search'](int x) { return true; },-10,-10) << "\n"; // infinite loop
```
Comment (frodakcin) [inline content/3_Silver/Binary_Search_Ans.mdx:717]: here

File: content/3_Silver/Binary_Search_Ans.mdx [hunk 8]
```diff
@@ -708,7 +723,7 @@ int main() {
 <JavaSection>
 
 ```java
-public static void main(String[] args) throws IOException
+public static void main(String['Binary Search'] args) throws IOException
```
Comment (frodakcin) [inline content/3_Silver/Binary_Search_Ans.mdx:726]: here

Comment (frodakcin) [reply content/3_Silver/Binary_Search_Ans.mdx:726]: I'm putting comments everywhere so if there's any that aren't "outdated" it's easy to tell if they aren't fisxed

File: content/3_Silver/Intro_Ordered.mdx [hunk 9]
```diff
@@ -66,20 +66,10 @@ export const problems = {
...
-			'contest/831/problem/C',
-			'Normal',
-			false,
-			[],
-			''
-		),
 		new Problem('Silver', 'Convention II', '859', 'Normal', true, [
-			'Sorting',
 			'Ordered Set',
 		]),
```
Comment (frodakcin) [inline content/3_Silver/Intro_Ordered.mdx:74]: Formatting

File: content/4_Gold/Sliding.mdx [hunk 10]
```diff
@@ -350,7 +365,7 @@ static ArrayList<Integer> maxSlidingWindow(int[] nums, int k) {
 	ArrayDeque<Integer> d = new ArrayDeque<Integer>();
 	for (int i = 0; i < k; i++) {
 		while(!d.isEmpty() && nums[i] > nums[d.peekLast()]) {
-    			d.pollLast();
+    		d.pollLast();
```
Comment (frodakcin) [inline content/4_Gold/Sliding.mdx:368]: Inconsistent space vs tabs?

**Review by frodakcin — CHANGES_REQUESTED** [review]
Minor issues, but I wrote a comment everywhere they occurred so there's a bunch of comments.

File: content/3_Silver/Binary_Search_Ans.mdx (hunk 6 above)
Comment (bqi343) [reply content/3_Silver/Binary_Search_Ans.mdx:667]: just commenting on this one time is enough :)

**Review by frodakcin — APPROVED** [review]
Looks good to me, but there's so many changes we're definitely gonna want more reviews.

File: content/3_Silver/Greedy.mdx [hunk 11]
```diff
@@ -102,29 +102,45 @@ export const problems = {
...
+		),
+		new Problem(
+			'CSA',
+			'Sure Bet',
+			'sure-bet',
 			'Normal',
 			false,
-			[],
-			'usaco-990'
+			['Greedy', 'Sorting', 'Two Pointers'],
```
Comment (frodakcin) [inline content/3_Silver/Greedy.mdx:123]: Should we stick to using "Two Pointers" or "2P" as the tag name?

**Review by frodakcin — COMMENTED** [review]
(not a full review)

File: content/3_Silver/Greedy.mdx (hunk 11 above)
Comment (bqi343) [reply content/3_Silver/Greedy.mdx:123]: let's go w/ 2P

**Review by frodakcin — APPROVED** [review]
Looks good.

File: content/3_Silver/Binary_Search.mdx [hunk 12]
```diff
@@ -0,0 +1,1047 @@
...
+			''
+		),
+		new Problem(
+			'Silver',
+			'Loan Repayment',
+			'991',
+			'Very Hard',
+			false,
+			['Binary Search', 'Sqrt'],
+			'requires some rather tricky analysis to speed up naive $mathcal{O}(Nlog N)$ solution'
```
Comment (dongliu0426) [inline content/3_Silver/Binary_Search.mdx:76]: \log

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: content/3_Silver/Binary_Search.mdx (hunk 12 above)
Comment (bqi343) [reply content/3_Silver/Binary_Search.mdx:76]: this doesn't actually appear anywhere so it doesn't matter :P

**Review by bqi343 — COMMENTED** [review]
@thecodingwizard is this ok

**Review by dongliu0426 — APPROVED** [review]

**Comment by bqi343** [comment]
Merging since it looks like redirects work as intended.

**Pull request merged**